*** local.c	2018-07-04 17:25:14.722841247 -0400
--- local.c	2018-07-13 20:32:30.282841247 -0400
***************
*** 20,25 ****
--- 20,27 ----
  #include "lock.h"
  #include "parse.h"
  
+ #include "hspace.h"
+ 
  extern HASHTAB htab_reserved_aliases;
  
  /* Called after all MUSH init is done.
***************
*** 38,44 ****
   * time. The latter two can be null pointers. The callback function returns true
   * if the softcode event should be triggered, false if it shouldn't.
   */
! #if 0 /* Change to 1 if you need local_timer functionality. */
    sq_register_loop(1, local_timer, NULL, NULL);
  #endif
  }
--- 40,47 ----
   * time. The latter two can be null pointers. The callback function returns true
   * if the softcode event should be triggered, false if it shouldn't.
   */
!   hs_init();
! #if 1 /* Change to 1 if you need local_timer functionality. */
    sq_register_loop(1, local_timer, NULL, NULL);
  #endif
  }
***************
*** 94,99 ****
--- 97,103 ----
  void
  local_dump_database(void)
  {
+   hs_dump();
  }
  
  /* Called when the MUSH is shutting down.
***************
*** 103,108 ****
--- 107,113 ----
  void
  local_shutdown(void)
  {
+   hs_shutdown();
  }
  
  /* Called when the MUSH is performing a dbck database check,
***************
*** 112,117 ****
--- 117,123 ----
  void
  local_dbck(void)
  {
+   hs_dump();
  }
  
  /* This is called exactly once a second after the MUSH has done all
***************
*** 121,127 ****
  bool
  local_timer(void *data __attribute__((__unused__)))
  {
! 
    /* The callback has to be set back up or it'll only run once. */
    return false;
  }
--- 127,133 ----
  bool
  local_timer(void *data __attribute__((__unused__)))
  {
!   hs_cycle();
    /* The callback has to be set back up or it'll only run once. */
    return false;
  }
*** flaglocal.dst	2018-08-01 19:50:24.564828035 -0400
--- flaglocal.c	2018-08-01 20:02:56.552991387 -0400
***************
*** 32,37 ****
--- 32,50 ----
  void
  local_flags(FLAGSPACE *flags __attribute__((__unused__)))
  {
+   add_flag("HS_ADMIN", '\0', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+   add_flag("HS_CELESTIAL", '\0', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+   add_flag("HS_COMM", '@', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+   add_flag("HS_COMPONENT", '&', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+   add_flag("HS_CONSOLE", '\0', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+   add_flag("HS_DRONE", '\0', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+   add_flag("HS_MISSION", 'y', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+   add_flag("HS_SHIP", '\0', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+   add_flag("HS_SHIPOBJ", '$', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+   add_flag("HS_SIM", '\0', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+   add_flag("HS_UNIVERSE", '\0', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+   add_flag("HS_WEAPON", '>', TYPE_PLAYER | TYPE_THING, F_WIZARD, F_WIZARD);
+ 
  #ifdef EXAMPLE
    if (strcmp(flags->name, "FLAG") == 0) {
      add_flag("BIG", 'B', TYPE_PLAYER | TYPE_THING, F_ANY, F_ANY);
*** cmdlocal.c	2018-07-04 17:23:25.470841247 -0400
--- cmdlocal.c	2018-07-13 20:30:37.418841247 -0400
***************
*** 20,25 ****
--- 20,27 ----
  #include "htab.h"
  #include "parse.h"
  
+ #include "hspace.h"
+ 
  extern HASHTAB htab_reserved_aliases;
  
  /* Called during the command init sequence before any commands are
***************
*** 80,83 ****
--- 82,95 ----
    command_add("@SILLY", CMD_T_ANY, "WIZARD ROYALTY", "SEE_ALL",
                "NOISY NOEVAL VERY", cmd_local_silly);
  #endif
+ #ifdef HSPACE
+   command_add("@SPACE", CMD_T_ANY | CMD_T_EQSPLIT, "WIZARD ROYALTY HS_SHIP HS_CONSOLE", NULL, "DUMP HALT LIST LOAD MOVE RECONFIGURE SPAWN START VERSION", cmd_space);
+   command_add("@CONSOLE", CMD_T_THING | CMD_T_EQSPLIT, NULL, NULL, "ABORT BURN CLOSEBAY COMBATSTAT COOLDOWN DOCK DUMP ESTATUS ETA EXTEND FIRE GATE HACK HEADING LAND LAUNCH LINK LOCK MANIFEST NAVMODE OPENBAY PRIMARY PROMPT PVPOFF PVPON QUICKSTAT RETRACT SCAN SECONDARY SHUNT SPEED SREP STANCE STATUS SYSSTAT TAXI TRACTOR UNLINK VIEW WAYPOINT", cmd_console);
+   command_add("BOARD", CMD_T_ANY | CMD_T_EQSPLIT, NULL, NULL, NULL, cmd_board);
+   command_add("COM", CMD_T_ANY | CMD_T_EQSPLIT, NULL, NULL, NULL, cmd_com);
+   command_add("DISEMBARK", CMD_T_ANY, NULL, NULL, NULL, cmd_disembark);
+   command_add("EJECT", CMD_T_ANY, NULL, NULL, NULL, cmd_eject);
+   command_add("MAN", CMD_T_ANY, NULL, NULL, NULL, cmd_man);
+   command_add("UNMAN", CMD_T_ANY, NULL, NULL, NULL, cmd_unman);
+ #endif
  }
*** funlocal.c	2018-07-04 17:23:25.486841247 -0400
--- funlocal.c	2018-07-13 20:30:37.418841247 -0400
***************
*** 22,27 ****
--- 22,29 ----
  #include "function.h"
  #include "parse.h"
  
+ #include "hspace.h"
+ 
  void local_functions(void);
  
  /* Here you can use the new add_function instead of hacking into function.c
***************
*** 39,42 ****
--- 41,54 ----
  #ifdef EXAMPLE
    function_add("SILLY", local_fun_silly, 1, 1, FN_REG);
  #endif
+ #if HSPACE
+   function_add("HS_CYCLE", fun_hs_cycle, 0, 0, FN_REG);
+   function_add("HS_SET", fun_hs_set, 2, 3, FN_WIZARD);
+   function_add("HS_GET", fun_hs_get, 2, 2, FN_REG);
+   function_add("HS_SREP", fun_hs_srep, 1, 1, FN_REG);
+   function_add("HS_COM", fun_hs_com, 1, 2, FN_REG);
+   function_add("HS_COOLDOWN", fun_hs_cooldown, 2, 2, FN_REG);
+   function_add("HS_LOAD", fun_hs_load, 1, 1, FN_WIZARD);
+   function_add("HS_SALVAGE", fun_hs_salvage, 2, 2, FN_WIZARD);
+ #endif
  }
*** Makefile	2018-08-01 19:40:35.246989869 -0400
--- Makefile	2018-07-16 17:11:28.588748292 -0400
***************
*** 7,14 ****
  OUTFILES=buildinf netmud info_slave ssl_slave
  
  # Libs EXCEPT for SQL ones
! LIBS=$(CLIBS)
! CFLAGS=$(CCFLAGS) $(SQL_CFLAGS)
  
  # On Windows, due to the bizarre way it handles cases in filesystems,
  # make thinks the file extension is .C and not .c and tries to compile files
--- 7,14 ----
  OUTFILES=buildinf netmud info_slave ssl_slave
  
  # Libs EXCEPT for SQL ones
! LIBS=$(CLIBS) space/libhspace.a
! CFLAGS=$(CCFLAGS) $(SQL_CFLAGS) -DHSPACE -Ispace
  
  # On Windows, due to the bizarre way it handles cases in filesystems,
  # make thinks the file extension is .C and not .c and tries to compile files
